controllers/datadogagent/component/agent/default.go

```diff
@@ -292,6 +294,7 @@ func volumeMountsForTraceAgent() []corev1.VolumeMount {
 		component.GetVolumeMountForAuth(),
 		component.GetVolumeMountForConfig(),
 		component.GetVolumeMountForDogstatsdSocket(true),
+		component.GetVolumeMountForRuntimeSocket(true),
```

Comment:
question: I thought that the live container feature should only affect the process container, why do we need the volume mount in the other containers too? Is it because of a different feature?

Reply:
In the operator v1 code, the runtime socket volume mount looked to be added to multiple containers by default: [core agent](https://github.com/DataDog/datadog-operator/blob/235ea457bc2b9e8fecceb28035b37a04eeccbecc/controllers/datadogagent/utils.go#L1442), [process agent](https://github.com/DataDog/datadog-operator/blob/235ea457bc2b9e8fecceb28035b37a04eeccbecc/controllers/datadogagent/utils.go#L1604), [security agent](https://github.com/DataDog/datadog-operator/blob/235ea457bc2b9e8fecceb28035b37a04eeccbecc/controllers/datadogagent/utils.go#L1788). In the helm chart, it was also added to the [trace agent container](https://github.com/DataDog/helm-charts/blob/20b27a7b408bf836d2d6fdf3d620f216fc021885/charts/datadog/templates/_container-trace-agent.yaml#L74), so I added the volume mount to all those containers to be safe. I didn't look into specific features to see whether the runtime socket was needed though